File: python/create_tempfile.py
import tempfile

# create an unlinked file object
tf = tempfile.TemporaryFile()

# create a linked file object with fixed pattern filename
tf = tempfile.NamedTemporaryFile(dir="/path/to/tmp", prefix="myfile", suffix=".dat")

# create a linked file object but it's not deleted after closed
tf = tempfile.NamedTemporaryFile(dir="/path/to/tmp", delete=False)

File: ruby/create_tempfile.rb
require "tempfile"

# create a linked file
tf = Tempfile.new("prefix")

# create a linked file with suffix
tf = Tempfile.new(["prefix", ".dat"])

# create a linked file in the specific directory
tf = Tempfile.new("prefix", "/path/to/dir")

# unlinked file is not supported explicitly
tf = Tempfile.new("prefix")
tf.unlink

